File: TypeScript/leetcode/longestCommonPrefix.ts
function longestCommonPrefix(strs: string[]): string {
  if (strs == null || strs.length === 0) {
    return "";
  }
  const shortest = strs.reduce((a, b) => (a.length < b.length ? a : b));
  for (let i = 0; i < shortest.length; i++) {
    const c = shortest.charAt(i);
    for (const other of strs) {
      if (other.charAt(i) !== c) {
        return shortest.substring(0, i);
      }
    }
  }
  return shortest;
}

File: Go/leetCode/LongestCommonPrefix.go
package leetcode

func LongestCommonPrefix(strs []string) string {
	if len(strs) == 0 {
		return ""
	}
	shortest := strs[0]
	for _, str := range strs {
		if len(str) < len(shortest) {
			shortest = str
		}
	}
	for i := 0; i < len(shortest); i++ {
		c := shortest[i]
		for _, str := range strs {
			if str[i] != c {
				return shortest[:i]
			}
		}
	}
	return shortest
}
